skimage/feature/_texture.pyx

```diff
@@ -264,3 +264,174 @@ def _local_binary_pattern(double[:, ::1] image,
                 output[r, c] = lbp
 
     return np.asarray(output)
+
+
+cdef inline Py_ssize_t _clip(Py_ssize_t x, Py_ssize_t low,
+                             Py_ssize_t high) nogil:
+    """Clips coordinate between high and low.
+
+    Parameters
+    ----------
+    x : int
+        Coordinate to be clipped.
+    low : int
+        The lower bound.
+    high : int
+        The higher bound.
+
+    Returns
+    -------
+    x : int
+        `x` clipped between `high` and `low`.
+    """
+
+    if(x > high):
+        return high
+    if(x < low):
+        return low
+    return x
+
+
+cdef inline cnp.double_t _integ(
+    cnp.double_t[:, ::1] img, Py_ssize_t r0, Py_ssize_t c0,
+        Py_ssize_t r1, Py_ssize_t c1) nogil:
+    """Integrate over the integral image in the given window
+
+    This method was created so that `multiblock_local_binary_pattern`
+    does not have to make a Python call.
+
+    Parameters
+    ----------
+    img : array
+        The integral image over which to integrate.
+    r0 : int
+        The row number of the top left corner.
+    c0 : int
+        The column number of the top left corner.
+    r1 : int
+        The row number of the bottom right corner.
+    c1 : int
+        The column number of the bottom right corner.
+
+    Returns
+    -------
+    ans : double
+        The integral over the given window.
+    """
+
+    r = _clip(r0, 0, img.shape[0] - 1)
+    c = _clip(c0, 0, img.shape[1] - 1)
+
+    r2 = _clip(r1, 0, img.shape[0] - 1)
+    c2 = _clip(c1, 0, img.shape[1] - 1)
+
+    cdef cnp.double_t ans = img[r1, c1]
+
+    if (r0 >= 1) and (c0 >= 1):
+        ans += img[r0 - 1, c0 - 1]
```

Comment:
@vighneshbirodkar This is the exact copy of
https://github.com/scikit-image/scikit-image/blob/master/skimage/transform/integral.py#L33